corehq/project_limits/models.py

```diff
@@ -80,3 +82,17 @@ class RateLimitedTwoFactorLog(models.Model):
     window = models.CharField(max_length=15, null=False)
     # largest input is 'number_rate_limited', 31 for headroom
     status = models.CharField(max_length=31, null=False)
+
+
+@audit_fields("limit")
+class SystemLimit(models.Model):
+    key = models.CharField(max_length=255)
+    limit = models.PositiveIntegerField()
+    # the domain field is reserved for extreme cases since limits should apply globally in steady state
+    domain = models.CharField(max_length=128, blank=True, default="")
+
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=['key', 'domain'], name='unique_key_per_domain_constraint')]
+
+    def __str__(self):
+        return f"{self.key}: {self.limit}"
```

Comment:
I think it'd be useful to add domain here.

Reply:
[c8dc267](https://github.com/dimagi/commcare-hq/pull/35675/commits/c8dc2676badd5a7c16465349ceabd069879e57e4)

Reply:
[19c4551](https://github.com/dimagi/commcare-hq/pull/35675/commits/19c45511e031a4a304a57c4d918989f0dbaa307b)